src/shared/index.ts

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ArtifactLocation, Log, Location, Region, Result } from 'sarif'
+
+type JsonLocation = { line: number, column: number } // Unused: pos
+type JsonRange = { value: JsonLocation, valueEnd: JsonLocation } // Unused: key, keyEnd
+export type JsonMap = Record<string, JsonRange>
+
+export type ResultId = [string, number, number]
+type _RegionBytes = [number, number] // byteOffset, byteLength
+type _RegionStartEndLineCol = [number, number, number, number] // start line, start col, end line, end col
+export type _Region
+	= number // single line
+	| _RegionBytes
+	| _RegionStartEndLineCol
+
+// Underscored members are optional in the source files, but required after preprocessing.
+declare module 'sarif' {
+	interface Log {
+		_uri: string
+		_uriUpgraded?: string
+		_jsonMap?: JsonMap
```

Comment:
>? [](start = 10, length = 1)

Per your comment above, this shouldn't be optional?

Reply:
Comment out of date after the recent enablement of Typescript Strict. Will update.

---
In reply to: [437555828](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r437555828) [](ancestors = 437555828)

Reply:
Updated the comment. Also added a bit more comments for some members.

---
In reply to: [437644402](https://github.com/microsoft/sarif-vscode-extension/pull/279#discussion_r437644402) [](ancestors = 437644402,437555828)